ui/src/callWithTelemetryAndErrorHandling.ts

```diff
@@ -69,6 +69,8 @@ export async function callWithTelemetryAndErrorHandling<T>(callbackId: string, c
 
 function handleError(context: IActionContext, callbackId: string, error: unknown): void {
     const errorData: IParsedError = parseError(error);
+    context = { ...context, ...errorData.actionContext };
```

Comment:
Don't think this is what you want, it will completely replace the telemetry and errorHandling objects if they exist on the errorData, for instance completely erasing any existing telemetry properties in the context.  And I'd suggest you need a test for this.

Reply:
I see. I though it will do a deep merge. Is there any other utility/pattern exist to merge the partial type other than checking each property?

Reply:
It might be easier if we change the type from `Partial<IActionContext>` to `(context: IActionContext) => void` so that people can edit the context themselves without us having to do a deep merge. Thoughts? Is that cumbersome?

To answer your original question - yes there are tools to do a deep merge, but I don't know offhand. Would have to look it up.

Reply:
Interesting to have a function.  Maybe not a bad idea.

You can search on npmjs.com for deep merge libraries.

Reply:
@EricJizbaMSFT, I didn't get your comment on change the type. Can you please elaborate on this?
yes there are several libraries to merge and I can use one, if we are going with that route.

Reply:
Not Applicable with the new design